freeipmi/freeipmi.go

```diff
@@ -196,15 +197,25 @@ func GetSensorData(ipmiOutput Result, excludeSensorIds []int64) ([]SensorData, e
 	return result, err
 }
 
+
 func GetCurrentPowerConsumption(ipmiOutput Result) (float64, error) {
 	if ipmiOutput.err != nil {
 		return -1, fmt.Errorf("%s: %s", ipmiOutput.err, ipmiOutput.output)
 	}
-	value, err := getValue(ipmiOutput.output, ipmiDCMICurrentPowerRegex)
+	// Check for Power Measurement are avail
+	value, err := getValue(ipmiOutput.output, ipmiDCMIPowerMeasurementRegex)
 	if err != nil {
 		return -1, err
 	}
-	return strconv.ParseFloat(value, 64)
+	// When Power Measurement in 'Active' state - we can get watts
+	if value == "Active" {
+		value, err := getValue(ipmiOutput.output, ipmiDCMICurrentPowerRegex)
+		if err != nil {
+			return -1, err
+		}
+		return strconv.ParseFloat(value, 64)
+	}
+	return -1, err
```

Comment:
err is always `nil` at this point, can we make that explicit by returning `-1, nil` here?

Reply:
Done